region/wreckedship/main/Bowling Alley.json

```diff
@@ -1003,10 +1014,55 @@
             "h_canShineChargeMaxRunway"
           ]}
         ]},
-        {"shinespark": {"frames": 40}}
+        {"or": [
+          {"shinespark": {"frames": 40, "excessFrames": 3}},
+          {"and": [
+            "canShinechargeMovement",
+            "canMidairShinespark",
+            {"shinespark": {"frames": 32, "excessFrames": 2}}
+          ]},
+          {"and": [
+            "canShinechargeMovementComplex",
+            "canMidairShinespark",
+            {"shinespark": {"frames": 27, "excessFrames": 2}}
+          ]},
+          {"and": [
+            "HiJump",
+            "canShinechargeMovementComplex",
+            "canMidairShinespark",
+            {"shinespark": {"frames": 18, "excessFrames": 2}}
+          ]}
+        ]}
       ],
       "clearsObstacles": ["A"]
     },
+    {
+      "link": [4, 5],
+      "name": "Use Flash Suit",
+      "requires": [
+        {"or": [
+          {"obstaclesCleared": ["A"]},
+          "h_canUsePowerBombs"
+        ]},
+        {"useFlashSuit": {}},
+        {"or": [
+          {"shinespark": {"frames": 32, "excessFrames": 3}},
+          {"and": [
+            "HiJump",
+            {"shinespark": {"frames": 29, "excessFrames": 3}}
+          ]},
+          {"and": [
+            {"shinespark": {"frames": 5, "excessFrames": 3}},
+            {"or": [
+              "canWalljump",
```

Comment:
harder-than-precise walljump (and consecutive) or phantoon not defeated.
Or if you can flash suit, you can also manage this?

Reply:
I think with flash suit this should be fine.